### DISCUSSION TITLE
Adios2 doesn't check for MPI Server/Client Support if not using MPICH

### DISCUSSION THREAD

**Discussion by RemiLacroix-IDRIS** [ornladios/ADIOS2#4201] [Q&A]
Hello,

Looking at the code, Adios2 doesn't check for MPI Server/Client Support if not using MPICH:
https://github.com/ornladios/ADIOS2/blob/8cd2d9481997a463e21d625f9f77abb343d69523/cmake/DetectOptions.cmake#L463-L492

Is this on purpose? Shouldn't newish version of OpenMPI work too?

Best regards,
Rémi

**Comment by eisenhauer** [comment]
Maybe?  Trying it and letting us know would expedite acceptance...  

@vicentebolea Any idea?

**Comment by vicentebolea** [comment]
OpenMPI support of MPI_connect/accept has been flaky if not broken in the past: https://github.com/open-mpi/ompi/issues/3458 https://github.com/open-mpi/ompi/pull/6446. We have observed that with particular backends selections and newer versions it does work, however, we have had a large amount of users complaining that the mpi dataset was not working in their builds with ompi, for that reason we decided to disable it by default when using ompi.

 You can force the check to be TRUE by setting the ADIOS2_HAVE_MPI_CLIENT_SERVER variable to ON/TRUE/1.
